src/test/java/org/broadinstitute/hellbender/engine/CRAMSupportIntegrationTest.java

```diff
@@ -41,6 +47,59 @@ public String getTestedClassName() {
         return testCases.toArray(new Object[][]{});
     }
 
+    @DataProvider(name = "cramFromPathIndex")
+    public Object[][] cramFromPathIndex() {
+        final File ref = new File(hg19MiniReference);
+        return new Object[][] {
+            // cram, index file, reference, intervals, expected read names
+            { new File(TEST_DATA_DIR, "cram_with_bai_index.cram"), new File(TEST_DATA_DIR, "cram_with_bai_index.cram.bai"),
+                    ref, Arrays.asList("1:1-200", "1:1000-2000"), Arrays.asList("a", "d", "e") },
+            { new File(TEST_DATA_DIR, "cram_with_crai_index.cram"), new File(TEST_DATA_DIR, "cram_with_crai_index.cram.crai"),
+                    ref, Arrays.asList("1:1000-2000", "3:300-400"), Arrays.asList("d", "e", "i", "j") }
+        };
+    }
+
+    @Test(dataProvider = "cramFromPathIndex")
+    public void testQueryCRAMWithIndexFromPath(
+            final File cramFile,
+            final File indexFile,
+            final File referenceFile,
+            final List<String> intervalArgs,
+            final List<String> expectedReadNames ) throws IOException {
+        final List<String> args = new ArrayList<>();
+        try (final FileSystem jimfs = Jimfs.newFileSystem(Configuration.unix())) {
+            final Path jimfsRootPath = jimfs.getRootDirectories().iterator().next();
+
+            // copy the cram, the index, the reference, the reference dictionary and the reference index to jimfs
+            final Path cramPath = Files.copy(cramFile.toPath(), jimfsRootPath.resolve(cramFile.getName()));
+            Files.copy(indexFile.toPath(), jimfsRootPath.resolve(indexFile.getName()));
+
+            final Path refPath = Files.copy(referenceFile.toPath(), jimfsRootPath.resolve(referenceFile.getName()));
+
+            final File referenceDict = ReferenceSequenceFileFactory.getDefaultDictionaryForReferenceSequence(referenceFile);
+            final Path referenceDictPath = referenceDict.toPath();
+            Files.copy(referenceDictPath, jimfsRootPath.resolve(referenceDict.getName()));
+
+            final Path referenceIndexPath = ReferenceSequenceFileFactory.getFastaIndexFileName(referenceFile.toPath());
+            Files.copy(referenceIndexPath, jimfsRootPath.resolve(referenceFile.getName() + ".fai"));
```

Comment:
Can you attempt to add functionality to `ArgumentsBuilder` (or a special `JimFSArgumentsBuilder` subclass) to handle the copying into JimFS and the translation into JimFS URIs?